src/tools/wpa/DataModel/QuicEtwEvent.cs

```diff
@@ -0,0 +1,21 @@
+﻿//
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+
+using System;
+using Microsoft.Diagnostics.Tracing;
+
+namespace MsQuicTracing.DataModel
+{
+    public class QuicEtwEvent : QuicEventBase
+    {
+#pragma warning disable CS8618
```

Comment:
We really should not be disabling this warning. I would much rather prefer make it nullable, and then check for null in the other 2 properties.

Reply:
```
    public class QuicEtwEvent : QuicEventBase
    {
        [DisallowNull]
        public TraceEvent? Event { get; set; }

        public override Guid Provider => Event?.ProviderGuid ?? throw new InvalidOperationException("Event must be set before calling provider");

        public override ushort ID => (ushort?)Event?.ID ?? throw new InvalidOperationException("Event must be set before calling ID");
    }
```

This is the class properly done to work correctly.

Reply:
We're going to end up with a lot of properties here. I really don't want to have to add a ton of nullable checks, both for code cleanliness and performance. A dozen or two extra null checks, per event, for 10 million events makes a difference.

Reply:
Performance is not a great argument, as the runtime will optimize a lot of that away correctly. And code cleanliness is still the worst argument ever. Correct code is ALWAYS better then clean code. Unconditionally.

Reply:
Once we have this all up and running, we can then try switching it out to this model to see how it performs. For now, I want to keep it as is.

Reply:
And I don't. Its wrong,.

Reply:
I understand. I completely disagree that performance is not a great argument. It is my primary goal (and concern) moving to this new plugin model. Things already take a considerable time with the old one for large files. Until we can prove something will not affect perf I will **always** go with the more performant design. As I said, we can revisit in the future.

Reply:
Then you will not be getting an approval from me.

Reply:
Slight refactor to eliminate the problem (at least at this layer).

Reply:
Ah. Thats a much better solution.